src/hooks/useFetchCompanyData.ts

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect } from 'react';
+
+import { CompanyData } from '../types/companyData';
+
+const apiKey = process.env.VITE_COMPANY_API_KEY;
+const objects = ['NVDA', 'AAPL', 'MSFT', 'AMZN', 'META', 'GOOG'];
+const cachedData = localStorage.getItem('cachedCompanyData');
+
+interface UseFetchDataReturn {
+	data: CompanyData[];
+	loading: boolean;
+}
+
+export default function useFetchCompanyData(): UseFetchDataReturn {
+	const [companyData, setCompanyData] = useState<CompanyData[]>([]);
+	const [loading, setLoading] = useState<boolean>(true);
+
+	const handleFetchSuccess = (data: CompanyData[]) => {
+		setCompanyData(data);
+		setLoading(false);
+	};
+
+	const handleFetchError = (error: Error | unknown) => {
+		console.error('Error fetching company data from server:', error);
+		// if fetch fails - due to the api calls limit, use data from localStorage
+		if (cachedData) {
+			try {
+				const parsedData = JSON.parse(cachedData);
+				setCompanyData(parsedData);
+				setLoading(false);
+			} catch (parseError) {
+				console.error('Error parsing cached data:', parseError);
+			}
+		} else {
+			console.error('No company cached data available.');
+			setLoading(false);
+		}
+	};
+
+	useEffect(() => {
+		const fetchData = async () => {
```

Comment:
isn't this function basically the same in all fetches? Maybe you can abstract it and use it everywhere?

Maybe this whole file could be simplified like this.